### DISCUSSION TITLE
Hyperparameter optimization - relationship with hyperparameter importance?

### DISCUSSION THREAD

**Discussion by jeekoong** [optuna/optuna#3892] [Q&A]
I am using Optuna for hyperparameter optimization of neural network.
There are 5 parameters:
dNum, un, fd : layers sizes of model / suggest_int 1 ~ 50
lr : learning rate / suggest_discrete_uniform 1e-7 ~ 1e-5
dc : decay / suggest_discrete_uniform 1e-7 ~ 1e-4

I use Cma-Es sampler instead of TPE, which does not consider relationship between parameters.
And I use wandb for callback.

After several times of retrieving results,
I found that there is some kind of relationship between importance and optimization.

Example 1
![ex1](https://user-images.githubusercontent.com/68321185/184605947-812f3847-0d64-4884-9b05-5b773e6413b8.png)
![ex11](https://user-images.githubusercontent.com/68321185/184605962-23becebd-0b2c-455f-a4f3-970594ed01d7.png)

Example 2
![ex2](https://user-images.githubusercontent.com/68321185/184605969-8f443d61-a5fd-420a-9ad1-28ac073e55b7.png)
![ex22](https://user-images.githubusercontent.com/68321185/184605974-4db29c4d-8fd0-4c92-a08e-336b210763d6.png)

As shown in Example 1 and 2,
all the parameters, except the one that has largest importance, tend to converge to the median of suggested range.
Even with similar importance results, this happens. I think this can cause issues.

I would like to know:

Why this happens?
If it is normal for this tendency to happen?
If I can change this by code modification
Thank you for reading my question.
